### PR TITLE
Fix for including today in Custom UI when option is checked with previous/next 1 day/week/month/year in date filter

### PR DESCRIPTION
**Pull request by cgopalan**
Fixes https://github.com/metabase/metabase/issues/8073

###### Before submitting the PR, please make sure you do the following 
-  [x] If there are changes to the backend codebase, run the backend tests with `lein test && lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter`
-  [x] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)

### CODE REVIEW

**Comment by cgopalan** [comment]
@camsaul @senior can you please review?

**Comment by cgopalan** [comment]
This is a bug fix. Anybody willing to review? @salsakran @camsaul @senior

**Comment by mazameli** [comment]
Verified this fixes #8073 in the UI. Could we get a code review from @senior?

Thanks for your contribution, @cgopalan!

**Comment by cgopalan** [comment]
@mazameli Thanks for the quick verification and for moving this along!

**Comment by cgopalan** [comment]
@mazameli seems like this code has been refactored in the latest master, and it seems like the refactoring incorporates these changes. Am I correct? If so, I will close this PR.

**Comment by cgopalan** [comment]
@mazameli @senior this PR is no longer needed, correct?

**Comment by mazameli** [comment]
Ah, that might be the case. @camsaul and @senior pretty extensively refactored the query processing code over the last few weeks, so it's certainly possible that this PR might be redundant now. Thanks for your contribution, though!

**Comment by MarkRx** [comment]
@cgopalan can you test the master branch to see if it still happens?

**Comment by cgopalan** [comment]
@MarkRx tested master and found that it still happens. Have added the fixes to the new refactored code now.
@mazameli @senior @camsaul can you please review this?

**Comment by cgopalan** [comment]
@mazameli @senior reminder :)

**Comment by MarkRx** [comment]
Please get this fixed in 0.31.

File: test/metabase/query_processor/middleware/desugar_test.clj [hunk 1]
```diff
@@ -43,6 +43,36 @@
...
+              :filter       [:between
+                             [:datetime-field [:field-id 1] :month]
+                             [:relative-datetime 0 :month]
+                             [:relative-datetime 1 :month]]}}
+  (desugar
+   {:database 1
+    :type     :query
+    :query    {:source-table 1
+               :filter       [:time-interval [:field-id 1] 1 :month {:include-current true}]}}))
+
```
Comment (senior) [inline test/metabase/query_processor/middleware/desugar_test.clj]: Need some tests here for the #{-1} clause.

**Review by senior — DISMISSED** [review]
Change looks good to me overall, just needs some more tests.

**Comment by cgopalan** [comment]
@senior Added tests. Please review.

File: test/metabase/query_processor/middleware/desugar_test.clj (hunk 1 above)
Comment (cgopalan) [reply test/metabase/query_processor/middleware/desugar_test.clj]: Done.

**Comment by cgopalan** [comment]
@senior reminder to review

**Pull request closed without merging**